pkg/devcontainer/config/metadata.go

```diff
@@ -12,3 +12,11 @@ type ImageMetadata struct {
 	DevContainerActions    `json:",inline"`
 	NonComposeBase         `json:",inline"`
 }
+
+func AddConfigToImageMetadata(config *DevContainerConfig, imageMetadataConfig *ImageMetadataConfig) {
+	userMetadata := &ImageMetadata{}
+	userMetadata.DevContainerConfigBase = config.DevContainerConfigBase
+	userMetadata.DevContainerActions = config.DevContainerActions
+	userMetadata.NonComposeBase = config.NonComposeBase
+	imageMetadataConfig.Config = append(imageMetadataConfig.Config, userMetadata)
```

Comment:
how does this work when merging 2 configs? If the base devcontainer.json has a value for `runArgs` and then the user file also has this, which one takes precedent?

Reply:
the merge occurs during function MergeConfiguration. This function only ease the build of the object it handle. The latest value in the array is the one taken into account